src/Surfnet/StepupMiddleware/MiddlewareBundle/Console/Command/BootstrapYubikeySecondFactorCommand.php

```diff
@@ -0,0 +1,148 @@
+use Surfnet\Stepup\Identity\Value\NameId;
+use Surfnet\StepupMiddleware\ApiBundle\Identity\Entity\UnverifiedSecondFactor;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Identity\Command\ProveYubikeyPossessionCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
+
+final class BootstrapYubikeySecondFactorCommand extends AbstractBootstrapCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setDescription('Creates a Yubikey second factor for a specified user')
+            ->addArgument('name-id', InputArgument::REQUIRED, 'The NameID of the identity to create')
+            ->addArgument('institution', InputArgument::REQUIRED, 'The institution of the identity to create')
+            ->addArgument(
+                'yubikey',
+                InputArgument::REQUIRED,
+                'The public ID of the Yubikey. Remove the last 32 characters of a Yubikey OTP to acquire this.'
+            )
+            ->addArgument(
+                'registration-status',
+                InputArgument::REQUIRED,
+                'Valid arguments: unverified, verified, vetted'
+            )
+            ->addArgument('actor-id', InputArgument::REQUIRED, 'The id of the vetting actor');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->tokenStorage->setToken(
+            new AnonymousToken('cli.bootstrap-yubikey-token', 'cli', ['ROLE_SS', 'ROLE_RA'])
+        );
+        $nameId = new NameId($input->getArgument('name-id'));
+        $institutionText = $input->getArgument('institution');
+        $institution = new Institution($institutionText);
+        $mailVerificationRequired = $this->requiresMailVerification($institutionText);
+        $registrationStatus = $input->getArgument('registration-status');
+        $yubikey = $input->getArgument('yubikey');
+        $actorId = $input->getArgument('actor-id');
+        $this->enrichEventMetadata($actorId);
+        if (!$this->tokenBootstrapService->hasIdentityWithNameIdAndInstitution($nameId, $institution)) {
+            $output->writeln(
+                sprintf(
+                    '<error>An identity with name ID "%s" from institution "%s" does not exist, create it first.</error>',
+                    $nameId->getNameId(),
+                    $institution->getInstitution()
+                )
+            );
+
+            return;
+        }
+        $identity = $this->tokenBootstrapService->findOneByNameIdAndInstitution($nameId, $institution);
+        $output->writeln(sprintf('<comment>Adding a %s Yubikey token for %s</comment>', $registrationStatus, $identity->commonName));
+        $this->beginTransaction();
+        $secondFactorId = Uuid::uuid4()->toString();
+
+        try {
+            switch ($registrationStatus) {
```

Comment:
I think it would be valuable to add a `default` with an exception, to catch wrong statuses that otherwise get unnoticed.

Reply:
Yes that wouild be a good idea, will open a PR to address this